src/key.rs

```diff
@@ -25,8 +25,11 @@ use Signing;
 use Verification;
 use constants;
 use ffi::{self, CPtr};
+#[cfg(feature = "zeroize_derive")]
+use zeroize::Zeroize;
 
 /// Secret 256-bit key used as `x` in an ECDSA signature
+#[cfg_attr(feature = "zeroize_derive", derive(Zeroize))]
```

Comment:
Given how trivial the generated implementation is, we might want to implement this ourselves so we don't add `syn` to this dependency tree? (Assuming none of the other dependencies already pull it in but I don't think so)

Reply:
Has the added bonus that we can use `zeroize` for the feature name :)

Reply:
On further thought I guess we don't even really need to feature guard it.

Reply:
Is 'no new dependencies' a hard rule @apoelstra ? I don't see how to do this without adding the dependency on `zeroize`.

Reply:
> Has the added bonus that we can use `zeroize` for the feature name :)

FWIW: You could have also achieved this by enabling the custom derive support through the feature that is exposed by `zeroize` instead of depending on `zeroize_derive` directly: https://docs.rs/zeroize/1.2.0/zeroize/#custom-derive-support

```toml
zeroize = { version = "1.2", default-features = false, optional = true, features = ["zeroize_derive"] }
```

This will re-export the `Zeroize` derive from the `zeroize` crate as `zeroize::Zeroize`. Serde uses the same pattern so you don't have to depend on `serde` and `serde_derive` at the same time.

Reply:
Cool, cheers man.